modules/microvm/virtualization/microvm/guivm.nix

```diff
@@ -227,20 +267,52 @@ in {
       };
     };
 
-    # Waypipe in GUIVM needs to communicate with AppVMs over VSOCK
-    # However, VSOCK does not support direct guest to guest communication
-    # The vsockproxy app is used on host as a bridge between AppVMs and GUIVM
-    # It listens for incoming connections from AppVMs and forwards data to GUIVM
-    systemd.services.vsockproxy = {
-      enable = true;
-      description = "vsockproxy";
-      serviceConfig = {
-        Type = "simple";
-        Restart = "always";
-        RestartSec = "1";
-        ExecStart = "${vsockproxy}/bin/vsockproxy ${toString cfg.waypipePort} ${toString cfg.vsockCID} ${toString cfg.waypipePort}";
+    systemd.services = {
+      # Waypipe in GUIVM needs to communicate with AppVMs over VSOCK
+      # However, VSOCK does not support direct guest to guest communication
+      # The vsockproxy app is used on host as a bridge between AppVMs and GUIVM
+      # It listens for incoming connections from AppVMs and forwards data to GUIVM
+      vsockproxy = {
+        enable = true;
+        description = "vsockproxy";
+        serviceConfig = {
+          Type = "simple";
+          Restart = "always";
+          RestartSec = "1";
+          ExecStart = "${vsockproxy}/bin/vsockproxy ${toString cfg.waypipePort} ${toString cfg.vsockCID} ${toString cfg.waypipePort}";
+        };
+        wantedBy = ["multi-user.target"];
       };
-      wantedBy = ["multi-user.target"];
+
+      ivshmemsrv = let
+        pidFilePath = "/tmp/ivshmem-server.pid";
+        ivShMemSrv = let
+          vectors = toString (2 * shmConfig.vmCount);
+        in
+          pkgs.writeShellScriptBin "ivshmemsrv" ''
+            chown microvm /dev/hugepages
+            chgrp kvm /dev/hugepages
+            if [ -S ${shmConfig.hostSocketPath} ]; then
+              echo Erasing ${shmConfig.hostSocketPath} ${pidFilePath}
+              rm -f ${shmConfig.hostSocketPath}
+            fi
+            ${pkgs.sudo}/sbin/sudo -u microvm -g kvm ${pkgs.qemu_kvm}/bin/ivshmem-server -p ${pidFilePath} -n ${vectors} -m /dev/hugepages/ -l ${(toString config.ghaf.profiles.applications.ivShMemServer.memSize) + "M"}
```

Comment:
Please avoid redefining the suffix `"M"` here.

Reply:
It's not a redefinition. I added a comment that defined shared memory size unit are MBytes:
https://github.com/jkuro-tii/ghaf/blob/4717c4e9df348dbde0e336a273365fa050daa440/modules/desktop/profiles/applications.nix#L27